x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/index/IndexCompatibility.java

```diff
@@ -27,9 +27,9 @@ public static Map<String, EsField> compatible(Map<String, EsField> mapping, Vers
             if (isPrimitive(dataType) == false) {
                 compatible(esField.getProperties(), version);
             } else if (isTypeSupportedInVersion(dataType, version) == false) {
-                EsField field = new UnsupportedEsField(entry.getKey(), dataType.name(), null, esField.getProperties());
+                EsField field = new UnsupportedEsField(entry.getKey(), dataType.nameUpper(), null, esField.getProperties());
                 entry.setValue(field);
-                propagateUnsupportedType(entry.getKey(), dataType.name(), esField.getProperties());
+                propagateUnsupportedType(entry.getKey(), dataType.nameUpper(), esField.getProperties());
```

Comment:
Is this meant to change?

Reply:
I think so?  This is the one tricky bit with this refactor, actually.  The old `DataType` class defined a `name` method, which is not the same as the enum name method, and you can't override enum's name.  You can see [here](https://github.com/elastic/elasticsearch/pull/109227/files#diff-0635d6d9ed5349bb1eeff14286503a11cce10e57bc4fc18f6ba2e35c0fa11bfeL49) where we generate the name as the upper case of either the type name or the es type (and I don't know why those are different; reconciling the two constructors for this class would be an excellent follow-up).  So when I had to rename the function, I thought `nameUpper` made some sense.

In a perfect world, the constant name, `typeName`, and `esType` would all be the same thing, and we could just use `DataType::name`, but that's not what it is right now.  It looks like the only ones where the results of `name` are different from the old results of `name` are `SOURCE`, `DOC_DATA_TYPE`, and `TSID_DATA_TYPE` (which would be `_SOURCE`, `_DOC`, and `_TSID`, respectively).  If we don't care about those, or don't mind renaming the constants, then we can replace this with the enum version of `name` and not stress it.

So unless something more subtle than my (currently pre-coffee) brain can pick up on is happening here, this is intentional, but for a bad reason.  I'm open to fixing it, but didn't want to just start changing behavior in what's supposed to be a pure refactor.

Reply:
gotcha